### PR TITLE
Added Forward, Backward and Central Difference Option.

### PR DESCRIPTION
**Pull request by jejjohnson**
**TLDR**: I added a forward, central, and backward difference option for all of the finite difference methods.

---

**Problem**:

There are many cases in PDEs where we need to use the backward difference method. For example, any advection PDE requires a backwards finite difference scheme otherwise the solution blows up (e.g. [12 steps NS](https://nbviewer.org/github/barbagroup/CFDPython/blob/master/lessons/07_Step_5.ipynb)). Currently in the package, the package only implements the central difference except in the special case where the central difference stencil was too small for the input size). However, the package is able to generate the offsets for all 3 schemes. I think having the option to choose which finite difference scheme would be extremely advantageous for all users. I myself needed it write away for some of the code I am using (e.g. [1d linear convection](https://github.com/jejjohnson/jaxsw/blob/eman/notebooks/dev/eman/12_steps_2_NS/1.2_linear_convection.ipynb))

---

**Solution**:

* I added the forward and the backward difference scheme to the `difference` function which is callable via a `method="central"|"forward"|"backward"`. I also added this option to all of the subsequent functions that depend on the difference function, i.e. the rest of them. 
* I also updated all of the tests to reflect the changes. They pass. I only had to change the tolerance for one of the tests.
* It is backwards compatible with the previous version because the default (`"central"`) is still the default for all methods.
* I fixed a few typos here and there.

---
**Potential Improvements**

* I didn't see a place to put this into the `fgrad` function. This might be helpful for some people in the future.
* I did not add an option to have different schemes for different dimensions, e.g. `grad`, `divergence`, etc. I can see this being potentially useful in the future.

### CODE REVIEW

**Comment by ASEM000** [comment]
Hello, thank you for your contribution.

I will try to merge As soon as possible.

p.s. for `fgrad`, you can use `offests = [-1,0]`  to define the backward difference.
I will try to improve the documentation of `fgrad` to reflect this piece of info.

**Comment by ASEM000** [comment]
@jejjohnson  Great work 🎉
